app/src/ui/add-repository/add-existing-repository.tsx

```diff
@@ -81,18 +103,39 @@ export class AddExistingRepository extends React.Component<IAddExistingRepositor
     this.checkIfPathIsRepository(path)
   }
 
+  private pathExists(directory: string): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      try {
+        FS.stat(directory, (err, exists) => {
+          if (err) {
+            resolve(false)
+          }
+
+          resolve(true)
+        })
+      } catch (e) {
+        resolve(false)
+      }
+    })
+  }
+
   private async checkIfPathIsRepository(path: string) {
-    this.setState({ path, isGitRepository: null })
+    this.setState({ path })
 
     const token = ++this.checkGitRepositoryToken
+    const pathExists = await this.pathExists(path)
+
+    let isRepo = false
 
-    const isRepo = await isGitRepository(this.resolvedPath(path))
+    if (pathExists) {
+      isRepo = await isGitRepository(this.resolvedPath(path))
+    }
 
     // Another path check was requested so don't update state based on the old
     // path.
     if (token !== this.checkGitRepositoryToken) { return }
 
-    this.setState({ path: this.state.path, isGitRepository: isRepo })
+    this.setState({ path: this.state.path, isGitRepository: isRepo, showNonGitRepositoryWarning: !isRepo })
```

Comment:
We can drop the `path: this.state.path` here, it's probably a remnant from before we used object spread